src/model/WaterHeaterMixed.hpp

```diff
@@ -364,6 +368,18 @@ namespace model {
 
     WaterHeaterSizing waterHeaterSizing() const;
 
+    boost::optional<PlantLoop> sourceSidePlantLoop() const;
+
+    boost::optional<PlantLoop> useSidePlantLoop() const;
+
+    bool removeFromSourceSidePlantLoop();
+
+    bool addToSourceSideNode(Node& node);
+
+    boost::optional<ModelObject> sourceSideInletModelObject() const;
+
+    boost::optional<ModelObject> sourceSideOutletModelObject() const;
```

Comment:
I'd also add aliases for useSideInlet/OutletModelObject.

(I also wonder if it wouldn't be more effective to provide `boost::optional<Node> sourceSideInletNode() const;` instead?)